lib/ansible/plugins/callback/mqtt.py

```diff
@@ -0,0 +1,332 @@
+        if not self.mqtt_hostname:
+            self._display.warning('MQTT_HOSTNAME environment variable is not '
+                                  'set this is required')
+            self.disabled = True
+
+    def _publish(self, topic, msg):
+        out_topic = self.base_topic + '/' + topic
+        msg['timestamp'] = datetime.datetime.utcnow().isoformat()
+        publish.single(out_topic, json.dumps(msg),
+                       hostname=self.mqtt_hostname, port=self.port,
+                       auth=self.auth, tls=self.tls,
+                       client_id=self.client_id)
+
+    def v2_playbook_on_play_start(self, play):
+        self.playbook = play.name
+        self.uuid = play._uuid
+        topic = 'playbooks/' + self.uuid + '/action/start'
+        msg = {
+            'status': "OK",
+            'host': self.hostname,
+            'session': self.session,
+            'playbook_name': self.playbook,
+            'playbook_id': self.uuid,
+            'ansible_type': 'start',
+        }
+        self._publish(topic, msg)
+
+    def v2_playbook_on_stats(self, stats):
+        """Display info about playbook statistics"""
+        hosts = sorted(stats.processed.keys())
+
+        for host in hosts:
+            stat = stats.summarize(host)
+            topic = 'playbook/' + self.uuid + '/stats/' + host
+            msg = {
+                'host': self.hostname,
+                'ansible_host': host,
+                'playbook_id': self.uuid,
+                'playbook_name': self.playbook,
+                'stats': stat
+            }
+            self._publish(topic, msg)
+        # Publish standalon status without stats
+        if self.errors > 0:
+            status = 'FAILED'
+        else:
+            status = 'OK'
+
+        topic = 'playbook/' + self.uuid + '/action/finish/' + status
+        msg = {
+            'playbook_id': self.uuid,
+            'playbook_name': self.playbook,
+            'status': status
+        }
+        self._publish(topic, msg)
+
+    def v2_runner_on_ok(self, result, **kwargs):
+        host = result._host.get_name()
+        topic = 'playbooks/' + self.uuid + '/tasks/' + host + '/OK'
+        data = {
+            'status': "OK",
+            'host': self.hostname,
+            'ansible_host': host,
+            'session': self.session,
+            'ansible_type': "task",
+            'playbook_name': self.playbook,
+            'playbook_id': self.uuid,
+            'ansible_task': str(result._task),
+            'ansible_result': self._dump_results(result._result)
+        }
+        self._publish(topic, data)
+
+    def v2_runner_on_failed(self, result, **kwargs):
+        host = result._host.get_name()
+        topic = 'playbooks/' + self.uuid + '/tasks/' + host + '/FAILED'
+        self.errors += 1
+        data = {
+            'status': "FAILED",
+            'host': self.hostname,
+            'ansible_host': host,
+            'playbook_id': self.session,
```

Comment:
Above `playbook_id` was set to `self.uuid`, but here it's being set to `self.session` which above is used for `session`. Maybe some copy/pasta got messed up?

Reply:
oops, yep that's a good catch. I'll respin this now